### PR TITLE
Replaced default browser intent with WebView

### PR DESCRIPTION
**Pull request by jaindiv26**
In this PR 
- I've replaced default browser intent with WebView which is much faster and reliable.
- Fixed creation of two identical accounts via login/pass.

### CODE REVIEW

File: app/src/main/java/org/xwiki/android/sync/activities/OIDC/OIDCActivity.kt [hunk 1]
```diff
@@ -92,12 +94,18 @@ class OIDCActivity: AppCompatActivity(), AccountClickListener {
...
         }
 
         init()
     }
 
     private fun init() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            CookieManager.getInstance().removeAllCookies(null)
+            CookieManager.getInstance().flush()
+        }
```
Comment (InsanusMokrassar) [inline app/src/main/java/org/xwiki/android/sync/activities/OIDC/OIDCActivity.kt:118]: So, what about Jelly Bean and KitKat

File: app/src/main/java/org/xwiki/android/sync/activities/OIDC/OIDCActivity.kt [hunk 2]
```diff
@@ -214,8 +223,21 @@ class OIDCActivity: AppCompatActivity(), AccountClickListener {
             selectedAccountName = accountName
             val flow: AuthorizationCodeFlow = createAuthorizationCodeFlow(accountName, serverUrl)
             flow.newAuthorizationUrl().setRedirectUri(REDIRECT_URI).build().let {
-                val browserIntent = Intent(Intent.ACTION_VIEW, Uri.parse(it))
-                startActivity(browserIntent)
+                runOnUiThread {
```
Comment (InsanusMokrassar) [inline app/src/main/java/org/xwiki/android/sync/activities/OIDC/OIDCActivity.kt]: Replace with `appCoroutineScope.launch(Dispatchers.Main)`

File: app/src/main/java/org/xwiki/android/sync/activities/OIDC/OIDCActivity.kt [hunk 3]
```diff
@@ -214,8 +223,21 @@ class OIDCActivity: AppCompatActivity(), AccountClickListener {
...
+                    binding.webview.webViewClient = object : WebViewClient() {
```
Comment (InsanusMokrassar) [inline app/src/main/java/org/xwiki/android/sync/activities/OIDC/OIDCActivity.kt]: Will be better to extract this to outside class

File: app/src/main/java/org/xwiki/android/sync/activities/SignInViewFlipper.kt [hunk 4]
```diff
@@ -183,7 +183,7 @@ class SignInViewFlipper(
         return appCoroutineScope.launch {
             val user = userAccountsRepo.createAccount(
                 UserAccount(
-                    accountName,
+                    "$accountName@temp",
```
Comment (InsanusMokrassar) [inline app/src/main/java/org/xwiki/android/sync/activities/SignInViewFlipper.kt:186]: What is `temp` and why it was added here? just to identify that this user is temporal?

**Review by InsanusMokrassar — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/xwiki/android/sync/activities/SignInViewFlipper.kt (hunk 4 above)
Comment (jaindiv26) [reply app/src/main/java/org/xwiki/android/sync/activities/SignInViewFlipper.kt:186]: Yes, so earlier two accounts are created simultaneously although one of them is intended to be deleted, it wasn't deleted. So by adding temp, we distinguish that this account is temporary and will be deleted now.

File: app/src/main/java/org/xwiki/android/sync/activities/OIDC/OIDCActivity.kt (hunk 2 above)
Comment (jaindiv26) [reply app/src/main/java/org/xwiki/android/sync/activities/OIDC/OIDCActivity.kt]: Done.

File: app/src/main/java/org/xwiki/android/sync/activities/OIDC/OIDCActivity.kt (hunk 1 above)
Comment (jaindiv26) [reply app/src/main/java/org/xwiki/android/sync/activities/OIDC/OIDCActivity.kt:118]: Done.

File: app/src/main/java/org/xwiki/android/sync/activities/OIDC/OIDCActivity.kt [hunk 5]
```diff
@@ -227,4 +247,26 @@ class OIDCActivity: AppCompatActivity(), AccountClickListener {
...
+            requestAccessToken(createAuthorizationCodeFlow(accountName, serverUrl), authorizationCode)
+        }
+    }
+}
+
+interface WebViewPageLoadedListener {
+    fun onPageLoaded(authorizationCode: String?, accountName: String)
+}
+
+class OIDCWebViewClient(private val webViewPageLoadedListener: WebViewPageLoadedListener,private val accountName: String) : WebViewClient() {
```
Comment (InsanusMokrassar) [inline app/src/main/java/org/xwiki/android/sync/activities/OIDC/OIDCActivity.kt]: Make it internal and extract to separated file

**Review by InsanusMokrassar — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/xwiki/android/sync/activities/OIDC/OIDCActivity.kt (hunk 5 above)
Comment (jaindiv26) [reply app/src/main/java/org/xwiki/android/sync/activities/OIDC/OIDCActivity.kt]: Done

File: app/src/main/java/org/xwiki/android/sync/activities/OIDC/OIDCActivity.kt (hunk 3 above)
Comment (jaindiv26) [reply app/src/main/java/org/xwiki/android/sync/activities/OIDC/OIDCActivity.kt]: Done

File: app/src/main/java/org/xwiki/android/sync/activities/OIDC/OIDCActivity.kt [hunk 6]
```diff
@@ -227,4 +248,14 @@ class OIDCActivity: AppCompatActivity(), AccountClickListener {
...
     }
+
+    override fun onPageLoaded(authorizationCode: String?, accountName: String) {
+        appCoroutineScope.launch {
+            requestAccessToken(createAuthorizationCodeFlow(accountName, serverUrl), authorizationCode)
+        }
+    }
+}
+
+interface WebViewPageLoadedListener {
```
Comment (InsanusMokrassar) [inline app/src/main/java/org/xwiki/android/sync/activities/OIDC/OIDCActivity.kt]: Must be part of OIDCWebViewClient.kt

**Review by InsanusMokrassar — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/xwiki/android/sync/activities/OIDC/OIDCActivity.kt (hunk 6 above)
Comment (jaindiv26) [reply app/src/main/java/org/xwiki/android/sync/activities/OIDC/OIDCActivity.kt]: Done

**Review by InsanusMokrassar — APPROVED** [review]

**Pull request merged**